### PR TITLE
Add toggle switch in ML and ML modal

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?
Add toggle switch in ML and ML modal to change between

### Related issue(s)/PR(s)
This PR replaces #14949 & #14895 due to branching issues

### To-Do
- [x] Add toggle switch to ML and ML modal
- [x] Update `localstorage` with new keys for ML view settings
- [x] Add required icons to DS and update dependencies
- [x] Add tests

### CODE REVIEW

**Comment by jhoward1994** [comment]
Unresolved comments copied from https://github.com/strapi/strapi/pull/14895. Apologies for the PR mess
@[gu-stav]I'd prefer if we'd prefix the key with STRAPI_
@[gu-stav]Also: could we build it in a way so that list is the default and we only persist if it was changed to grid?

@[jhoward1994]should list be the default view? seeing as today we only have the grid view?

File: packages/core/upload/admin/src/pages/App/tests/MediaLibrary.test.js [hunk 1]
```diff
@@ -493,5 +500,20 @@ describe('Media library homepage', () => {
...
       expect(screen.queryByText('header.actions.add-assets')).not.toBeInTheDocument();
     });
+
+    it('displays the appropriate switch to change the view', () => {
+      const setView = jest.fn();
+      usePersistentState.mockReturnValueOnce([viewOptions.GRID, setView]);
+      renderML();
+
+      const listSwitch = screen.queryByTestId('switch-to-list-view');
+      const gridSwitch = screen.queryByTestId('switch-to-grid-view');
```
Comment (ronronscelestes) [inline packages/core/upload/admin/src/pages/App/tests/MediaLibrary.test.js:510]: same as for `BrowseStep` :)

**Review by ronronscelestes — APPROVED** [review]
thank you for the fixes 💅
I don't know how the debate ended about list view or grid view by default but I agree that since ML has always been grid view, that might be confusing for people to have list view as default - I think it should be a product question we could ask?

**Comment by jhoward1994** [comment]
> thank you for the fixes 💅 I don't know how the debate ended about list view or grid view by default but I agree that since ML has always been grid view, that might be confusing for people to have list view as default - I think it should be a product question we could ask?

I think it makes sense that the grid view is the default but I will ask the question in the JIRA ticket

I think @gu-stav was also thinking that we could build this in a way that only persists the value to localstorage once the value has been changed.

I'm reluctant to modify `usePersistentState` to cover this one case and would rather use this hook as it is a consistent way to interact with localstorage across the app. WDYT?

**Comment by ronronscelestes** [comment]
> I think @gu-stav was also thinking that we could build this in a way that only persists the value to localstorage once the value has been changed.
> 
> I'm reluctant to modify `usePersistentState` to cover this one case and would rather use this hook as it is a consistent way to interact with localstorage across the app. WDYT?

mmh I see - it's a good question
we would need to check everywhere it is used, if it would break anything to refactor (shouldn't but 🤷🏻‍♀️) + it's an `helper-plugin` hooks which means other people can use it in their project

but I would say if it's something that we want to do anyway, it shouldn't be done in this feature but should its own refactor, wdyt?

**Pull request merged**